### DISCUSSION TITLE
Is strapi supported to work on Azure app service on linux having a xenial kernel

### DISCUSSION THREAD

**Discussion by deepakmoses** [strapi/strapi#7237] [Legacy Topics]
Is strapi supported to work on Azure app service on linux having a xenial kernel to connect to mongo db 4.2 atlas database? The kernel i see in the app service on linux is  4.4.0.128 generic . Is this supported?

**Comment by lauriejim** [comment]
Ping @derrickmehaffy to help you on this one :)

**Reply by deepakmoses** [reply]
Thank you @lauriejim 
@derrickmehaffy  we ran into serious issues of strapi not connecting to the cosmos db via mongo api. As this is not a supported configuration by strapi we are looking at options to have the strapi which is currently on a linux app service to connect to mongo atlas 4.2. Strapi documentation mentions ubuntu 18.04 in the list of supported operating systems to connect to mongo atlas 4.2. However going by the mongo documentation links it says ubuntu 16.0.4 lts is also supported. However there is one issue and we would like to get a confirmation whether it is supported config. Azure app service is not pure ubuntu . The  cat /etc/issue shows 
Debian GNU/Linux 9 \  . However the kernel is ubuntu 4.4.0-128-generic #154-Ubuntu SMP Fri May 25 14:15:18 UTC 2018 x86_64 GNU/Linux . Can this be considered equivalent of ubuntu 16.0,4 LTS? Is this os  supported with strapi to make database connections with mongo atlas 4.2? Appreciate your response and thanks for your support.

**Reply by derrickmehaffy** [reply]
One of the key things I really didn't care for with the App services was how the files were loaded in (when deploying from your local system) in which it more or less creates a zip of your local project directory (including node_modules) and dumps that to the app service then extracts it.

When it comes to linux Debian is more or less the same as Ubuntu with the main difference is Ubuntu is typically a bit more "bleeding edge" meaning some of the dependencies are more up to date than typical Debian. Rarely do I see Debian used in production environments and it's usually Ubuntu, CentOS/RHEL but you shouldn't have any issues with Debian directly.

When it comes to the Distro IE 20.04 vs 18.04 vs 16.04. The short and sweet of that is we test on 18.04 and not 16.04 anymore and we support based on what we test on. A key thing to note is that 16.04 is almost 5 years old now (quick tip on ubuntu version numbers xx.yy where xx is the year it was released and yy was the month so 16.04 was released in April 2016)

MongoDB atlas is supported but something I would suggest is possibly looking into Azure's PostgreSQL or MySQL databases.
